includes/utils/class-amp-validation-utils.php

```diff
@@ -733,6 +896,81 @@ public static function locate_sources( DOMNode $node ) {
 				$sources[] = $parsed_comment['source'];
 			}
 		}
+
+		$is_enqueued_link = (
+			$node instanceof DOMElement
+			&&
+			'link' === $node->nodeName
+			&&
+			preg_match( '/(?P<handle>.+)-css$/', (string) $node->getAttribute( 'id' ), $matches )
+			&&
+			isset( self::$enqueued_style_sources[ $matches['handle'] ] )
+		);
+		if ( $is_enqueued_link ) {
+			$sources = array_merge(
+				self::$enqueued_style_sources[ $matches['handle'] ],
+				$sources
+			);
+		}
+
+		/**
+		 * Script dependency.
+		 *
+		 * @var _WP_Dependency $script_dependency
+		 */
+		if ( $node instanceof DOMElement && 'script' === $node->nodeName ) {
+			$enqueued_script_handles = array_intersect( wp_scripts()->done, array_keys( self::$enqueued_script_sources ) );
+
+			if ( $node->hasAttribute( 'src' ) ) {
+
+				// External script.
+				$src = $node->getAttribute( 'src' );
+				foreach ( $enqueued_script_handles as $enqueued_script_handle ) {
+					$script_dependency  = wp_scripts()->registered[ $enqueued_script_handle ];
+					$is_matching_script = (
+						$script_dependency
+						&&
+						$script_dependency->src
+						&&
+						// Script attribute is haystack because includes protocol and may include query args (like ver).
+						false !== strpos( $src, preg_replace( '#^https?:(?=//)#', '', $script_dependency->src ) )
+					);
+					if ( $is_matching_script ) {
+						$sources = array_merge(
+							self::$enqueued_script_sources[ $enqueued_script_handle ],
```

Comment:
The `$enqueued_script_handle` also needs to be passed to the validation error to include as the `handle`.

Reply:
Done in 3c0d724